tempesta_fw/sock.c

```diff
@@ -552,10 +552,15 @@ __ss_close(struct sock *sk, int flags)
 	 * the queued work is closing and simply pretend that socket closing
 	 * event happened before the socket transmission event.
 	 *
-	 * The socket is owned by current CPU, so don't need to check its
-	 * liveness.
+	 * The socket is owned by current CPU, so there's no need to check
+	 * if it's live. However, in some cases this may be called multiple
+	 * times on the same socket. Do it only once for the socket.
 	 */
 	bh_lock_sock(sk);
+	if (unlikely(!ss_sock_live(sk))) {
+		bh_unlock_sock(sk);
+		return SS_OK;
+	}
```

Comment:
No, this is slightly different, and it's in a different place than in that patch. The comment above this code block explains why that was introduced. Also, please see [this comment in http.c](https://github.com/tempesta-tech/tempesta/pull/660/files#r101928722).